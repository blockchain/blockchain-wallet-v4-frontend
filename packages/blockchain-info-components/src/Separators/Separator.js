import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
`
const Content = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  white-space: nowrap;
  padding-left: ${props =>
    props.align === 'left' || props.align === 'center' ? '10px' : '0'};
  padding-right: ${props =>
    props.align === 'right' || props.align === 'center' ? '10px' : '0'};
`
const BaseSeparator = styled.div`
  width: 100%;
  height: 1px;
  background-color: ${props => props.theme['gray-2']};
  margin: ${props => (props.margin ? props.margin : '10px 0')};
`

const Separator = props => {
  const { children, align, className, margin } = props

  return children ? (
    <Wrapper className={className}>
      {align !== 'right' && <BaseSeparator margin={margin} />}
      <Content align={align}>{children}</Content>
      {align !== 'left' && <BaseSeparator margin={margin} />}
    </Wrapper>
  ) : (
    <BaseSeparator className={className} margin={margin} />
  )
}

Separator.propTypes = {
  align: PropTypes.oneOf(['left', 'right', 'center'])
}

Separator.defaultProps = {
  align: 'center'
}

export default Separator
