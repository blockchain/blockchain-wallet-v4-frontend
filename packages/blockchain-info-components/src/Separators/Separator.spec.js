import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Separator from './Separator'

describe('Separator component', () => {
  it('default renders correctly', () => {
    const component = shallow(<Separator align='right' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
