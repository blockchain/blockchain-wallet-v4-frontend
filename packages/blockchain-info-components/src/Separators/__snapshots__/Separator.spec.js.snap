// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator component default renders correctly 1`] = `<Separator__BaseSeparator />`;
